File: backend/models/promotion.py
from sqlalchemy import Column, Integer, String, Date, Float, Index
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import relationship

# Epic Title: Develop Frontend Interface for Promotions

Base = declarative_base()

class Promotion(Base):
    __tablename__ = 'promotions'
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    code = Column(String(20), nullable=False, unique=True)
    description = Column(String(255), nullable=False)
    discount_amount = Column(Float, nullable=False)
    expiration_date = Column(Date, nullable=False)
    
    discounts = relationship("Discount", back_populates="promotion")
    
    __table_args__ = (
        Index('ix_code', 'code'),
    )

File: backend/models/promotion.js
const { DataTypes, Model } = require('sequelize');
const sequelize = require('../services/database');

// Epic Title: Integrate Promotion System with Order Management

class Promotion extends Model {}

Promotion.init({
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  code: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true,
  },
  description: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  discount_amount: {
    type: DataTypes.FLOAT,
    allowNull: false,
  },
  expiration_date: {
    type: DataTypes.DATE,
    allowNull: false,
  },
}, {
  sequelize,
  modelName: 'Promotion',
});

module.exports = Promotion;